ls1intum/Artemis
# `Development`: Add helper script to convert Spring YAML configuration files to their equivalent ENV format

### Checklist
#### General



### Motivation and Context
Spring configurations can be given either as YAML files, properties files, or as environment variables due to [the configuration binding](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-Configuration-Binding).
For Docker/Kubernetes setups it might be easier to use the ENV-variable option. 

### Description

To simplify the conversion from one setup type to the other, this script automates the conversion of the configuration from YAML to ENV.

The script is hosted in https://github.com/b-fein/spring-yaml-to-env including some tests and linting set up. The changes to Artemis itself are only the addition of a short explanation in the documentation.

### Steps for Testing
1. Try to convert your configuration files using the script.
2. The script should not print any warnings or errors to the console.
3. (Optional, might be trickier to test):
    - If you have got a Systemd-Service that starts Artemis, you can define an `EnvironmentFile=…` there that contains the content of the script output. Artemis should then still start when the `yml` configs are removed.
    - If you have got a Docker setup, you can use the environment variables there as well.
    - **NOTE**: This does not work currently if you have defined custom Docker images for the CI-system runners (options `artemis.continuous-integration.build.images.…`). This is not due to the script, but due to how Spring converts the ENV-variable names back into lowercase. The Artemis code that handles the loading in this case cannot handle the image names being in lowercase. ⇒ Fix in #5596 

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
unchanged
